### PR TITLE
Maps are now placed in the center of the Z level

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

Z levels are 255 by 255 tiles, but only ~240 by 240 area is accessible because the margins are reserved for smooth transition between space levels.
The levels were loaded with x and y offsets set as 1, regardless of the level bounds.
So when you load a custom DMM that doesn't take into account these smooth transition margins, its western and southern part was cut.

This PR makes it so that stations are always placed in the center of the level.

This has no effect of the 255 by 255 maps, but smaller maps, like runtime station or custom ruins loaded as stations, will not require to have the smooth transition margin tiles reserved in DMM. And they will not end up having parts in the inaccessible area.

## Why It's Good For The Game

You can now properly load any space ruin as a custom map without it being fucked up by map cropping.

## Changelog
:cl:
fix: The small maps are now placed in the center of Z level to avoid having parts in the inaccessible area.
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
Please point me to the a define that says that z-levels bounds are 255x255 if we have it.

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -257,7 +257,10 @@ Used by the AI doomsday and the self-destruct nuke.
 	// load the maps
 	for (var/P in parsed_maps)
 		var/datum/parsed_map/pm = P
-		if (!pm.load(1, 1, start_z + parsed_maps[P], no_changeturf = TRUE))
+		var/bounds = pm.bounds
+		var/x_offset = bounds ? round(255 / 2 - bounds[MAP_MAXX] / 2) : 0
```
Comment (Fikou) [inline code/controllers/subsystem/mapping.dm:261]: dont use hardcoded 255, use maxx and maxy

Comment (MTandi) [reply code/controllers/subsystem/mapping.dm:261]: I didn't find the define. Can you please point at it?

Comment (AffectedArc07) [reply code/controllers/subsystem/mapping.dm:261]: `world.maxx` and `world.maxy`

**Comment by Mothblocks** [comment]
Tests failing

**Comment by MTandi** [comment]
Researching shuttles

**Comment by MTandi** [comment]
@Mothblocks Checked all shuttles, everything is good now. Thank you.

**Pull request closed without merging**